mne/utils/check.py

```diff
@@ -562,6 +562,40 @@ def _check_depth(depth, kind='depth_mne'):
     return _handle_default(kind, depth)
 
 
+def _check_dict_keys(user_dict, valid_keys,
+                     dict_name="Channel name(s)", valid_name="info"):
+    """Check that the keys in dictionary are valid against a set list.
+
+    Return the input dictionary if it is valid,
+    otherwise raise a ValueError with a readable error message.
+
+    Parameters
+    ----------
+    user_dict : dict
+        The name of the parameter to check. This is used in the error message.
+    valid_keys : list
+        All possible valid key names.
+
+    Raises
+    ------
+    ValueError
+        When the key of the dict is not one of the valid options.
+
+    Returns
+    -------
+    user_dict
+        When the keys are deemed acceptable the dictionary is returned.
+    """
+    sorted_dict = sorted(list(user_dict))
```

Comment:
Not a great variable name choice here, as it's no longer a dict.

Also this might be cleaner / faster to read if this line and next were done using `set()`

Reply:
I thought about `set` but it shouldn't be needed since keys of dicts should be unique anyway

Reply:
True, but I thought `set(user_dict) - set(valid_keys)` would be cleaner than these 2 lines. Not a blocker though, just a nitpick. I feel that `set()` is underutilized 😆

Reply:
Ahh yes that would work and I agree it's cleaner

Reply:
I would also use the set difference.

Reply:
Using set now, thanks for the tip, I didnt know that one.